service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/support/ShareCaseControllerTest.java

```diff
@@ -50,51 +64,136 @@
     @Autowired
     private MockMvc mockMvc;
 
-    @Test
-    @WithMockUser(authorities = "caseworker-publiclaw-systemupdate")
-    void shouldShareCaseWithUsers() throws Exception {
-        final String caseId = UUID.randomUUID().toString();
-        final String user1 = UUID.randomUUID().toString();
-        final String user2 = UUID.randomUUID().toString();
+    @Captor
+    private ArgumentCaptor<String> queryCaptor;
 
-        given(authTokenGenerator.generate()).willReturn(SERVICE_AUTH_TOKEN);
+    @Nested
+    class ShareAllCasesFromGivenLocalAuthority {
 
-        final MvcResult result = shareCase(caseId, List.of(user1, user2));
+        @BeforeEach
+        void init() {
+            Mockito.reset(caseUser, coreCaseDataService, authTokenGenerator);
+        }
 
-        assertThat(result.getResponse().getStatus()).isEqualTo(SC_OK);
-        verifyAccessGranted(caseId, user1);
-        verifyAccessGranted(caseId, user2);
-    }
+        @Test
+        @WithMockUser(authorities = "caseworker-publiclaw-systemupdate")
+        void shouldShareCases() throws Exception {
+            final String localAuthority = "LA1";
+            final String localAuthority2 = "LA2";
+            final String user1 = UUID.randomUUID().toString();
+            final String user2 = UUID.randomUUID().toString();
+
+            CaseDetails case1 = caseDetails(localAuthority);
+            CaseDetails case2 = caseDetails(localAuthority2);
+            CaseDetails case3 = caseDetails(null);
+            CaseDetails case4 = caseDetails("");
+            CaseDetails case5 = caseDetails(localAuthority);
+
+            given(authTokenGenerator.generate()).willReturn(SERVICE_AUTH_TOKEN);
+            given(coreCaseDataService.searchCases(any(), queryCaptor.capture()))
+                .willReturn(List.of(case1, case2, case3, case4, case5));
```

Comment:
should case 2 be returned here?

Reply:
it is to test the additional check, changed a bit test data to make it more realistic